calling-apis/post-requests-with-axios.md

```diff
@@ -0,0 +1,221 @@
+
+1. What is the API call that lists all existing Trip packages?
+2. How many existing trips are there?
+3. What is the API call that creates a new Trip package?
+4. What is the verb and URL of that request?
+5. What are the required query parameters? What are the optional query parameters?
+
+<br/>
+
+<details style="max-width: 700px; margin: auto;">
+
+  <summary>
+    Here is what Matteo discovers with the documentation and Postman.
+  </summary>
+
+1. `get` `https://trektravel.herokuapp.com/trips`
+1. Depends, but Matteo will note the shape of the expected JSON response
+1. `post` `https://trektravel.herokuapp.com/trips`
+1. `post`, and `https://trektravel.herokuapp.com/trips`
+1. Required: `name`, `continent`, `category`, `weeks`, and `cost`. Optional: `about`
+
+</details>
+
+## Making a `POST` request
+
+The syntax for making a POST request using `axios` is very similar to making a GET request, but we will use the `post` function instead.
+
+### An Aside: Working With Ambiguous Documentation
+
+What happens if Matteo [finds that the official `POST` request documentation is ambiguous](https://axios-http.com/docs/post_example), and he can't find more satisfactory resources?
+
+For example, Matteo could go to the POST request page and see this code, with zero explanation:
+
+<!-- prettier-ignore-start -->
+```js
+axios.post('/user', {
+    firstName: 'Fred',
+    lastName: 'Flintstone'
+  })
+  .then(function (response) {
+      console.log(response);
+  })
+  .catch(function (error) {
+      console.log(error);
+  });
+```
+<!-- prettier-ignore-end -->
+
+The best thing for Matteo to do in this situation is:
+
+1. Make hypotheses, using context clues and what he knows about APIs
+1. Test them
+
+From this example code, we can see that the second argument for `axios.post` is this object literal:
+
+<!-- prettier-ignore-start -->
+```js
+{
+    firstName: 'Fred',
+    lastName: 'Flintstone'
+}
+```
+<!-- prettier-ignore-end -->
+
+Matteo sees that there are two keys, `firstName` and `lastName`. From his knowledge of APIs, these keys don't look like they're describing parts of an HTTP request like HTTP headers or params. Additionally, this example request is being made to `/user`.
+
+Matteo hypothesizes that this object literal holds the HTTP request body data, and he can test his hypothesis against his API.
+
+### Sending a Request to the Ada Trip Reserving Service™ Trip API™™™
+
+Matteo is ready to try to make a `POST` request to the back-end service. He [determines from the API documentation](https://github.com/AdaGold/trip-api) that the request he wants to make is:
+
+- `POST` request to `https://trektravel.herokuapp.com/trips`
+- Request Body:
+  - `name`
+  - `continent`
+  - `about`
+  - `category`
+  - `weeks`
+  - `cost`
+
+The `post` function takes in two parameters: URL, and an optional object. This optional object represents the HTTP request body data that we want to send.
```

Comment:
the existing phrasing sounds definitive, but we are continuing on from Matteo's own tests, so let the phrasing reflect this
```suggestion
From the limited `axios` documentation, he has determined that the `post` function takes in two parameters: a URL, and an object. This object represents the HTTP request body data that we want to send.
```